openfl-workspace/tf_3dunet_brats/README.md

```diff
@@ -0,0 +1,148 @@
+
+```bash
+user@localhost ~$ tree ${DATA_PATH} -L 2
+${DATA_PATH}/MICCAI_BraTS2020_TrainingData
+├── BraTS20_Training_001
+│   ├── BraTS20_Training_001_flair.nii.gz    <── The MRI FLAIR channel (best one for prediction)
+│   ├── BraTS20_Training_001_seg.nii.gz      <── The ground truth label
+│   ├── BraTS20_Training_001_t1.nii.gz       <── The T1-weighted MRI channel
+│   ├── BraTS20_Training_001_t1ce.nii.gz     <── The T1-Contrast Enhanced-weighted MRI channel
+│   └── BraTS20_Training_001_t2.nii.gz       <── The T2-weighted MRI channel
+├── BraTS20_Training_002
+│   ├── BraTS20_Training_002_flair.nii.gz
+│   ├── BraTS20_Training_002_seg.nii.gz
+│   ├── BraTS20_Training_002_t1.nii.gz
+│   ├── BraTS20_Training_002_t1ce.nii.gz
+│   └── BraTS20_Training_002_t2.nii.gz
+├── ...
+├── BraTS20_Training_369
+│   ├── BraTS20_Training_369_flair.nii.gz
+│   ├── BraTS20_Training_369_seg.nii.gz
+│   ├── BraTS20_Training_369_t1.nii.gz
+│   ├── BraTS20_Training_369_t1ce.nii.gz
+│   └── BraTS20_Training_369_t2.nii.gz
+├── name_mapping.csv
+└── survival_info.csv
+```
+If `tree` is not installed, then run `sudo apt-get install tree` to install it (Ubuntu).
+
+3. In order for each collaborator to use separate slice of data, we split main folder into subfolders, one for each collaborator. **NOTE:** In the real world each collaborator will have it's own data and this split already exists. We're splitting here to simulate a federation with different participants.
+
+#### `split_directory.sh`
+```bash 
+#!/bin/bash
+# Split the BraTS data directory into NUM_COLLABORATORS
+
+SOURCE=${1}  # The directory where the BraTS dataset is located (e.g. ~/data/MICCAI_BraTS2020_TrainingData)
+DESTINATION=${2}   # The destination directory for the randomized, split training data folders
+NUM_COLLABORATORS=${3:-2}  # The number of collaborator splits for the subdirectories
+
+help() {
+    echo
+    echo "======================================================================="
+    echo "~$ split_directory.sh BRATS_DATA_SOURCE_DIRECTORY DESTINATION_DIRECTORY"
+    echo "======================================================================="
+    echo
+    echo "BRATS_DATA_SOURCE_DIRECTORY: The directory where the BraTS dataset is located (e.g. ~/data/MICCAI_BraTS2020_TrainingData)"
+    echo "DESTINATION DIRECTORY: The destination directory for the randomized, split training data folders (e.g. ~/brats_data_split)"
+    echo "NUM_COLLABORATORS: The number of collaborator splits for the subdirectories (default: 2)"
+    echo "-h, --help            display this help and exit"
+    echo
+    echo
+}
+
+if [ "$#" -lt 2 ] || ! [ -d ${1} ]; then
+    help
+    exit 1
+fi
+
+get_seeded_random()
+{
+  seed="$1"
+  openssl enc -aes-256-ctr -pass pass:"$seed" -nosalt \
+    </dev/zero 2>/dev/null
+}
+
+# Remove the destination directory if it exists
+rm -r ${DESTINATION}
```

Comment:
Here should be a check for directory existence in order to avoid the following error:
`rm: cannot remove '~/data': No such file or directory`

Reply:
I added a check to see if the directory exists.